lib/blocs/create_snapshot/create_snapshot_cubit.dart

```diff
@@ -0,0 +1,183 @@
+    if (_isValidHeightRange()) {
+      // TODO -
+    }
+
+    // declares the writing stream to the temporal file
+    final tempFile = File(_tempFilePath);
+    final tempFileSink = tempFile.openWrite();
+
+    emit(ComputingSnapshotData(
+      driveId: driveId,
+      range: range,
+    ));
+
+    // declare the GQL read stream out of arweave
+    final gqlEdgesStream = _arweave.getSegmentedTransactionsFromDrive(
+      driveId,
+      minBlockHeight: range.start,
+      maxBlockHeight: range.end,
+    );
+
+    // transforms the stream of arrays into a flat stream
+    final flatGQLEdgesStream = gqlEdgesStream.expand((element) => element);
+
+    // maps the items to GQL Nodes
+    final gqlNodesStream = flatGQLEdgesStream.map((edge) => edge.node);
+
+    // declares the reading stream from the SnapshotItemToBeCreated
+    final snapshotItemToBeCreated = SnapshotItemToBeCreated(
+      blockStart: 0, // TODO: get from the range (?)
+      blockEnd: currentHeight, // TODO: get from the range (?)
+      driveId: driveId,
+      subRanges: HeightRange(rangeSegments: [range]),
+      source: gqlNodesStream,
+      jsonMetadataOfTxId: jsonMetadataOfTxId,
+    );
+
+    // Stream snapshot data to the temporal file
+    await for (final item in snapshotItemToBeCreated.getSnapshotData()) {
+      tempFileSink.add(item);
+    }
+
+    // close the file as writing, and open as reading
+    await tempFileSink.close();
+    final tempFileRead = tempFile.openRead();
+
+    // declares the new snapshot entity
+    final snapshotEntity = SnapshotEntity(
+      id: const Uuid().v4(), // TODO: take from arguments (?)
+      driveId: driveId,
+      blockStart: range.start,
+      blockEnd: range.end,
+      dataStart: range.start,
+      dataEnd: range.end,
+      data: Uint8List.fromList(
+        await tempFileRead.expand((element) => element).toList(),
+      ),
+    );
+
+    final profile = _profileCubit.state as ProfileLoggedIn;
+    final wallet = profile.wallet;
+    final preparedTx = await _arweave.prepareEntityTx(
+      snapshotEntity,
+      wallet,
+      // key is null because we don't re-encrypt the snapshot data
+    );
+
+    snapshotEntity.txId = preparedTx.id;
+
+    final uploadSnapshotItemParams = CreateSnapshotParameters(
+      signedTx: preparedTx,
+      addSnapshotItemToDatabase: () => _driveDao.transaction(() async {
+        await _driveDao.writeSnapshotEntity(snapshotEntity);
+      }),
+    );
+
+    // TODO: estimate upload cost
+    const arUploadCost = '0';
+    const usdUploadCost = 0.0;
+
+    // TODO: emit ConfirmUpload or ComputeSnapshotDataFailure
+    emit(ConfirmUpload(
+      snapshotSize: tempFile.lengthSync(),
+      arUploadCost: arUploadCost,
+      usdUploadCost: usdUploadCost,
+      createSnapshotParams: uploadSnapshotItemParams,
+    ));
+  }
+
+  bool _isValidHeightRange() {
+    return range.end <= currentHeight;
+  }
+
+  Future<String> jsonMetadataOfTxId(String txId) async {
```

Comment:
Nit: make it private
```suggestion
  Future<String> _jsonMetadataOfTxId(String txId) async {
```

Reply:
cb3c9b7877656fcb2476025e1ee14a6a94112608